src/main/webapp/src/app/hidden-gem.service.ts

```diff
@@ -20,4 +20,18 @@ export class HiddenGemService {
   handleError(error: HttpErrorResponse) {
     return throwError('A data error occured, please try again.');
   }
+
+  findHiddenGemReccomendation(data: any) : Observable<HiddenGem[]> {
+    const params = new HttpParams().append("preferences", data);
+    return this.httpClient
+      .get<HiddenGem[]>('/recommendation', {params})
```

Comment:
I'm still trying to understand the code, so I might be wrong, but are these params used? If no, maybe add a comment somewhere as a TODO for later, like where these params will be used, etc. If you think that isn't necessary, feel free to ignore this comment

Reply:
Done. Added in TODO comments. The function will probably have to change again once again when we start sending user's data from the form for filtering in the request.